### ISSUE TITLE
Brave Adblocker should be allowed to run in brave:// or chrome:// URLs

### ISSUE THREAD

**Issue by ghost** [brave/brave-browser#34287]
## Description 

There is a new nice Brave Player feature implemented by @sangwoo108 https://github.com/brave/brave-core/pull/20730, the problem is while it is a nice feature to have, it runs in the `brave://player/youtube/*` URL, which means the adblocker is not allowed to run there.

This means there is not much advantage on watching the video in Brave Player, because you will get all connections through that  would have gotten blocked by the adblocker in Youtube website or an embedded Youtube video, making watching videos in normal Youtube '_more private_'.

These are the XHRs rules that aren't getting blocked:
```
||youtube-nocookie.com/api/stats/qoe?
||play.google.com/log
```
Found in EasyPrivacy [easyprivacy_specific.txt#L1372](https://github.com/easylist/easylist/blob/db15a0175329556ca42cf2d907f46ac547d3d1fc/easyprivacy/easyprivacy_specific.txt#L1372) and [easyprivacy_specific.txt#L870](https://github.com/easylist/easylist/blob/db15a0175329556ca42cf2d907f46ac547d3d1fc/easyprivacy/easyprivacy_specific.txt#L870) espectively.

Just like there is a flag `brave://flags/#brave-extension-network-blocking` that shows the power of the adblocker to run and block anywhere in Brave without any restrictions, there should be a way to also block these trackers or _unnecessary network request_ from Youtube, at least in the Brave Player and any future implementation if it uses any brave/chrome URL, to get the same adblocking experience than the website would have or any embedded Youtube video and make the feature good in terms of blocking trackers.

## Steps to Reproduce

   1. Enable `brave://flags/#brave-player`
   2. go to `brave://player/youtube/UyT8CmrrObY`
   3. Brave adblocker is disabled like any other brave:// or chrome:// URL which means every tracker will get through.

## Expected result:
Get trackers blocked where it is needed, even in `brave://` URLs

## Brave version (brave://version info)

1.62.45 Chromium: 119.0.6045.134 (Official Build) nightly (64-bit)

**Comment by diracdeltas** [comment]
Pete can confirm but I think this is working as intended. If we do want to allow adblocking extensions, we should allow it just for the chrome-untrusted player origin, not the top level brave frame.

**Comment by diracdeltas** [comment]
nvm i just re-read the spec and it seems the adblock scripts should be running in this frame. cc @sangwoo108

**Comment by pes10k** [comment]
Yes, just as @diracdeltas notes, the main goal of the brave-viewer feature is to continue applying Brave's adblocking and network filtering (i.e., adblock-rs) policies in the embedded youtube frame. The primarily goal of the feature is to continue the existing blocking policies in the frame, but in a way that makes it impossible for YouTube to connect back to the user's account (since the brave viewer frame is loaded in a sandboxed/emphemeral storage area)

**Comment by pes10k** [comment]
thanks @Emi-Emi-Emi , this is a good catch.  We should make sure we're running adblock-rs policies in the brave-viewer frame.  I want to emphasize that the privacy risk is very low here (the youtube embed is loaded from an empty storage area, so even if logging network requests get fired, they can't be attached to your Google/YT profile).

That said, we should make sure we're applying network-filtering policies in Brave-viewer.  Tagging @antonok-edm too

**Comment by stoletheminerals** [comment]
>how can Brave adblocker be controlled in such circumstances?

Should we have a Shields button enabled for some brave:// URLs, like in this case where we load untrusted content? wdyt @pes10k?

**Comment by pes10k** [comment]
i think it'd be nice to have a special shields setting here (we wouldn't want to treat all of brave:// the way we're treating brave-viewer).  But I also think thats not something we need for a MVP/v1 if its a non-trivial addition

**Comment by diracdeltas** [comment]
> i think it'd be nice to have a special shields setting here (we wouldn't want to treat all of brave:// the way we're treating brave-viewer). But I also think thats not something we need for a MVP/v1 if its a non-trivial addition

just want to reiterate i don't think we should apply any shields to brave:// origins. this needs to only apply to the youtube iframe.

**Comment by bsclifton** [comment]
Closing as we are not doing additional work on brave://player at this time
